compiler/rustc_trait_selection/src/traits/project.rs

```diff
@@ -408,6 +409,11 @@ impl<'a, 'b, 'tcx> TypeFolder<'tcx> for AssocTypeNormalizer<'a, 'b, 'tcx> {
             constant.eval(self.selcx.tcx(), self.param_env)
         }
     }
+
+    fn fold_constant(&mut self, constant: mir::ConstantKind<'tcx>) -> mir::ConstantKind<'tcx> {
+        let constant = constant.super_fold_with(self);
+        constant.eval(self.selcx.tcx(), self.param_env)
```

Comment:
```suggestion
        bug!("AssocTypeNormalizer should not be called on mir bodies");
```
at least that's my assumption. If not, isn't `super_fold_with` incorrect here as we eval the inner `ty::Const` which requires it to be representable as a `ValTree`